pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -255,12 +256,65 @@ func configFromEnv() (cfg Config, ok bool) {
 	return
 }
 
+// Allows Setting up of openstack credentials by reading clouds.yaml
+//
+// Looks for clouds.yaml file in the following paths in order:
+// 1. OS_CLIENT_CONFIG_FILE
+// 2. Current directory.
+// 3. unix-specific user_config_dir (~/.config/openstack/clouds.yaml)
+// 4. unix-specific site_config_dir (/etc/openstack/clouds.yaml)
+func configFromYAML() (cfg Config, err error) {
+	clientOpts := new(clientconfig.ClientOpts)
+	cloud, err := clientconfig.GetCloudFromYAML(clientOpts)
+
+	// Exit as a failure if unable to read clouds.yaml
+	if err != nil {
+		fmt.Println(err)
+		return cfg, err
+	}
+
+	cfg.Global.AuthURL = cloud.AuthInfo.AuthURL
```

Comment:
I think this is a good time to refactor the [`Config`](https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L154)  struct. We could replace the current Global definition with a field that holds a `clientconfig.Cloud` object.

Reply:
A new commit sounds good. I'd have it in this PR, tho.